interview-questions.md

```diff
@@ -7,44 +7,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
-
-  Researched answer:
-
+  Your answer: Get and Delete are two RESTful routes that must always be passed params.
 
+  Researched answer: The RESTful routes which must always be passed params are update, destroy, show, and edit. This is because update needs an ID to specify which instance is being referenced so that it can make sure it's updating the correct entry. The destroy route also needs an ID in order to know which instance the user wants to delete. Similarly, show also needs an ID because it is being used to display a specific entry in the table. Edit needs an ID to specify which post is being edited. This information gets passed to the edit form which is different than new because the user wants to modify existing information, not create a new instance of student.
 
 3. Name three rails generator commands. What is created by each?
 
-  Your answer:
+  Your answer: Rails generate Model, creates a new model. Rails generate Migration creates/updates the migration file. Generate controller generates a new Controller.
 
-  Researched answer:
+  Researched answer: The three rails generator commands are generate controller, generate model, and generate resource. Generate controller creates a new controller, a helper method, a routes, and a view. Generate model creates a model within a given database, and also creates a table. The table will use lowercase letters and given a plural name based on the name of the model. An Animal model will create a "animals" table. The third rails generator command is generate resource which creates embedded code and acts as a template for the RESTful routes. This saves the user time by not having to manually create each individual route. An interesting tidbit that I learned while researching generate commands is that if you don't drop the built-in test code when initially creating a rails app, creating a controller will also generate a test.rb file for the controller itself.
 
+4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
 
+action: "GET"    route: /students
 
-4. Consider the Rails routes below. What is the name of the controller method that would be called by each route? What action would each of the controller methods perform?
+  The controller method for this route is index. Index is used to retrieve all entries in the students table.
 
-action: "GET"    route: /students          
+action: "POST"   route: /students
 
-action: "POST"   route: /students       
+  The controller method here would be create. Create would take the information stored in a new webpage and use it to create a new student.
 
 action: "GET"    route: /students/new
 
-action: "GET"    route: /students/2  
+  The controller method here would be new. New creates a page where a user could input information, which would then be used by a create method to create a new student.
```

Comment:
Good concept. With web development, verbiage creates cohesive and better understanding in conversations. Especially with troubleshooting. The new controller returns a form (or page with a form). Concept is spot on with the rest of your answer.